producer.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+# thoth-graph-refresh-job-producer
+# Copyright(C) 2020 Sai Sankar Gochhayat, Fridolin Pokorny
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Refresh data stored in the graph database.
+
+Produces messages regarding package depending on the component in which is used,
+which are picked up by investigator to schedule solvers.
+"""
+
+import logging
+import random
+import os
+import asyncio
+
+from thoth.common import init_logging
+from thoth.common import OpenShift
+from prometheus_client import CollectorRegistry, Gauge
+
+from thoth.messaging import MessageBase
+from thoth.messaging.unresolved_package import UnresolvedPackageMessage
+from thoth.messaging.unrevsolved_package import UnrevsolvedPackageMessage
+from thoth.common import __version__ as __common__version__
+from thoth.storages import __version__ as __storage__version__
+from thoth.storages import GraphDatabase
+
+__version__ = "0.1.1"
+__service_version__ = (
+    f"{__version__}+storage.{__storage__version__}.common.{__common__version__}"
+)
+
+_LOGGER = logging.getLogger("thoth.graph_refresh_job")
+
+app = MessageBase().app
+
+init_logging()
+_GRAPH_DB = GraphDatabase()
+_GRAPH_DB.connect()
+
+_OPENSHIFT = OpenShift()
+
+prometheus_registry = CollectorRegistry()
+THOTH_MY_NAMESPACE = os.getenv("NAMESPACE", "thoth-test-core")
+
+# Metrics Exporter Metrics
+_METRIC_INFO = Gauge(
+    "thoth_graph_refresh_job_info",
+    "Thoth Graph Refresh Job information",
+    ["env", "version"],
+    registry=prometheus_registry,
+)
+_METRIC_INFO.labels(THOTH_MY_NAMESPACE, __service_version__).inc()
+
+
+@app.command()
+async def main() -> None:
+    """Schedule refresh messages for packages that are not yet analyzed by solver."""
```

Comment:
Produce Kafka messages depending on the knowledge that needs to be acquired for a certain package.